### ISSUE TITLE
enableFreeze causes useSharedValue's value to reset between screens in bridgeless mode

### ISSUE THREAD

**Issue by vilindberg** [software-mansion/react-native-screens#2533]
### Description

I originally created this issue in the react-native-reanimated repo (https://github.com/software-mansion/react-native-reanimated/issues/6728) but maybe this repo is a better fit for it.

**The problem:**
When having enableFreeze(true), useSharedValue resets to the initial value when moving to another screen within a bottom tab navigator and then back again. Since FirstScreen is not unmounted, I would expect it to keep the value at 300.

It seems to work (staying at 300) in RN 0.73 and in 0.74 when I disable new architecture, so it was likely introduced by the bridgeless mode.

https://github.com/user-attachments/assets/5fc9c205-37d2-4604-8520-665ede5dd106

### Steps to reproduce

1. Run the provided sample repo (https://github.com/vilindberg/reanimated-shared-value-between-stacks)
2. Wait for the width to animate to 300 on FirstScreen
3. Navigate to SecondScreen
4. Navigate back to FirstScreen and the width will have been reset to 100

### Snack or a link to a repository

https://github.com/vilindberg/reanimated-shared-value-between-stacks

### Screens version

4.3.0

### React Native version

0.76.2

### Platforms

iOS

### JavaScript runtime

None

### Workflow

None

### Architecture

Fabric (New Architecture)

### Build type

None

### Device

iOS simulator

### Acknowledgements

Yes

**Comment by YasinAkimura** [comment]
Ah, I can confirm I happened to get the same bug when switching tabs with freeze enabled but I ignored it as I didn't need the value and I was just playing around. But funny thing is when doing an update it pulls back the shared value so it's not actually lost.

**Comment by vilindberg** [comment]
@YasinAkimura Yeah, I experienced the same thing. After the next re-render of that screen the updated shared value is used. It's like the freezed state doesn't track shared value updates.

**Comment by alany411** [comment]
I can also confirm that this is happening when the new architecture is enabled with `enableFreeze(true)`.

**Comment by vilindberg** [comment]
Looks like this issue got solved in react-native-reanimated [3.17.2](https://github.com/software-mansion/react-native-reanimated/releases/tag/3.17.2). My guess would be that this lovely PR solve it, [#7316](https://github.com/software-mansion/react-native-reanimated/pull/7316).
